content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,430 @@
+
+### Removal of array shortcuts
+
+Prisma version 5.0.0 drops support for a number of array shortcuts. These shortcuts were a way to add a single element as a value to an array-based operator instead of an array of one element. In an effort to make our typings more consistent, we now require array values for these operators.
+
+In most cases, the fix will be as simple as wrapping the existing value in an array. Below are some specific examples of needed changes in Prisma 5.
+
+#### <inlinecode>OR</inlinecode> operators
+
+The following code in Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  OR: { email: 'foo@example.com' },
+})
+```
+
+Will need to be changed to the following in Prisma 5:
+
+```js highlight=2;edit
+prisma.user.findMany({
+  OR: [{ email: 'foo@example.com' }],
+})
+```
+
+`OR` operators will only accept array values.
+
+#### <inlinecode>in</inlinecode> and <inlinecode>notIn</inlinecode> operators
+
+Similar to `OR`, `in` and `notIn` require array values.
+
+Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  id: { in: 123 },
+})
+
+prisma.user.findMany({
+  id: { notIn: 123 },
+})
+```
+
+Prisma 5:
+
+```js highlight=3,9;edit
+prisma.user.findMany({
+  id: {
+    in: [123],
+  },
+})
+
+prisma.user.findMany({
+  id: {
+    notIn: [123],
+  },
+})
+```
+
+If your `in` and `notIn` values are only one element, you can also update your code to not use these operators:
+
+```js highlight=2,6;normal
+prisma.user.findMany({
+  id: 123,
+})
+
+prisma.user.findMany({
+  id: { not: 123 },
+})
+```
+
+#### <inlinecode>path</inlinecode> argument for filtering on JSON fields in Postgres
+
+[When filtering on JSON fields in a Postgres table](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-a-json-field) the `path` argument now only accepts an array.
+
+When using the following schema:
+
+```prisma
+datasource db {
+  provider = "postgresql"
+}
+
+model User {
+  id       String @id
+  settings Json
+}
+```
+
+Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  where: {
+    settings: {
+      path: 'someSetting',
+      equals: someValue,
+    },
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=4;edit
+prisma.user.findMany({
+  where: {
+    settings: {
+      path: ['someSetting'],
+      equals: someValue,
+    },
+  },
+})
+```
+
+<Admonition type="info">
+
+Note: This change only affects PostgreSQL databases. MySQL databases are not affected.
+
+</Admonition>
+
+#### Composite and Scalar lists
+
+Inputs for operations on lists of [Composite](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-composite-types) and [Scalar](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#scalar-fields) types now only accept array values. The following examples all use the following schema:
+
+```prisma
```

Comment:
The example above also lists the `datasource`. Either this one should as well, or both can drop it (which I would prefer).